src/model.js

```diff
@@ -24,29 +32,44 @@ function Model () {}
  * @param {function} callback
  */
 Model.prototype.getData = function (req, callback) {
-  let filename = `${req.params.id}.geojson`
-  fs.readFile(`${dataDir}/${filename}`, (err, dataBuffer) => {
+  const filename = `${req.params.id}.geojson`;
+  const filePath = `${this.dataDirPath}/${filename}`
+  fs.readFile(filePath, (err, dataBuffer) => {
     if (err && err.errno === -2) {
       err.code = 404
-      err.message = `File "${filename}" not found.`
+      err.message = 'File not found'
+      console.log(`${err.message}: ${filePath}`)
     } else if (err) {
       err.code = 500
     }
-    if (err) return callback(err)
 
-    // translate the response into geojson
-    const geojsonStr = dataBuffer.toString()
-    const geojson = translate(JSON.parse(geojsonStr))
+    if (err) {
+      return callback(err)
+    }
 
-    // Cache data for 10 seconds at a time by setting the ttl or "Time to Live"
-    geojson.ttl = 10
+    try {
+      // translate the response into geojson
+      const geojsonStr = dataBuffer.toString()
+      const geojsonParsed = JSON.parse(geojsonStr)
+      const geojson = translate(geojsonParsed)
 
-    // Add metadata
-    geojson.metadata = geojson.metadata || {}
-    geojson.metadata.title = 'Koop GeoJSON'
-    geojson.metadata.name = filename
-    geojson.metadata.description = `GeoJSON from ${filename}`
-    callback(null, geojson)
+      // Cache data for 10 seconds at a time by setting the ttl or "Time to Live"
+      geojson.ttl = geojsonParsed.metadata || 10
+
+      const detectedGeometryType = geojson.metadata.geometryType
+      // Add metadata
+      geojson.metadata = geojsonParsed.metadata || {}
+      geojson.metadata.geometryType = detectedGeometryType
+      geojson.metadata.title = (geojsonParsed.metadata && geojsonParsed.metadata.title) || 'Koop GeoJSON'
+      geojson.metadata.name = (geojsonParsed.metadata && geojsonParsed.metadata.name) || filename
+      geojson.metadata.description = (geojsonParsed.metadata && geojsonParsed.metadata.description) || `GeoJSON from ${filename}`
```

Comment:
Use lodash's `_.get` with a default on these metadata assignments - it's already used in koop and will make these assignments more readable.

Reply:
Lodash was not used in the original project. Happy to bring it in.